src/greynir_topic/model.py

```diff
@@ -337,3 +347,51 @@ def topic_vector(self, lemmas: List[LemmaString]) -> TopicVector:
     def similarity(topic_vector_a: TopicVector, topic_vector_b: TopicVector) -> float:
         """ Return the cosine similarity of two sparse topic vectors """
         return matutils.cossim(topic_vector_a, topic_vector_b)
+
+
+    def calculate_similarity_index(self) -> None:
+        """ Transform corpus to LSI space and index it """
+        corpus_tfidf = self.load_tfidf_corpus()
+        if self._dictionary is None:
+            self.load_dictionary()
+        assert self._dictionary is not None
+        if self._model is None:
+            self.load_lsi_model()
+        assert self._model is not None
+
+        # Calculate the similarity index
+        simindex = similarities.Similarity(
+            self.simindex_filename,
+            self._model[corpus_tfidf],
+            num_features=len(self._dictionary)
+        )
+
+        # Save the similarity index
+        simindex.save(self.simindex_filename)
+        self._simindex = simindex
+
+    def load_similarity_index(self) -> None:
+        """ Load similarity index to local variable """
+        self._simindex = similarities.Similarity.load(self.simindex_filename)
+
+    def train_similarity(
+        self, corpus: Corpus, *,
+        keep_temp_files: bool = False,
+        min_count: int = 3, max_ratio: float = 0.5
+    ) -> None:
+        """ Train the model for similarity calculations """
+        self.train(corpus, keep_temp_files=True, min_count=min_count, max_ratio=max_ratio)
+        self.calculate_similarity_index()
+        if not keep_temp_files:
+            self.remove_temp_files()
+
+    def nn(self, topic_vector: TopicVector, num_neighbors: int = None) -> list:
+        """ Return a list of indexes for the items in corpus that are most similar to given topic vector. """
+        if self._simindex is None:
+            self.load_similarity_index()
+        assert self._simindex is not None
+
+        if num_neighbors:
+            return [k for k, i in sorted(enumerate(self._simindex[topic_vector]), key=lambda i: -i[1])][:num_neighbors]
+        else:
+            return [k for k, i in sorted(enumerate(self._simindex[topic_vector]), key=lambda i: -i[1])]
```

Comment:
Better:
```
   # Obtain a list of similarities, with one float for each document in the corpus
   similarities = self._simindex[topic_vector]
   # Sort into descending order by similarity, maintaining the document indices
   neighbors = [ix for ix, _ in sorted(enumerate(similarities), key=lambda t: -t[1])]
   return neighbors[:num_neighbors] if num_neighbors else neighbors
```

Reply:
...also, it might be useful to have a similarity cutoff threshold, e.g. don't return documents with similarity below 0.9.